frontend/packages/console-telemetry-plugin/src/listeners/segment.ts

```diff
@@ -111,37 +111,57 @@ export const eventListener: TelemetryEventListener = async (
   properties?: any,
 ) => {
   if (TELEMETRY_DISABLED || !apiKey) {
-    // eslint-disable-next-line no-console
-    console.debug(
-      'console-telemetry-plugin: telemetry disabled - ignoring telemetry event:',
-      eventType,
-      properties,
-    );
-    return;
-  }
-  if (TELEMETRY_DEBUG) {
-    // eslint-disable-next-line no-console
-    console.debug('console-telemetry-plugin: received telemetry event:', eventType, properties);
+    if (TELEMETRY_DEBUG) {
+      // eslint-disable-next-line no-console
+      console.debug(
+        'console-telemetry-plugin: telemetry disabled - ignoring telemetry event:',
+        eventType,
+        properties,
+      );
+    }
     return;
   }
   switch (eventType) {
     case 'identify':
       {
         const { user, ...otherProperties } = properties;
-        const id = user?.metadata?.name;
-        if (id) {
+        const clusterId = otherProperties?.clusterId;
+        const organizationId = otherProperties?.organizationId;
+        const username = user?.metadata?.name;
```

Comment:
```suggestion
        const username = user?.username;
```
if this user is fetched from redux we need to use new UserInfo API

Reply:
This code is just refactored, we used that since 4.11, but I will check it on cluster.

https://github.com/openshift/console/blob/release-4.11/frontend/packages/console-telemetry-plugin/src/listeners/segment.ts#L85
...
https://github.com/openshift/console/blob/release-4.15/frontend/packages/console-telemetry-plugin/src/listeners/segment.ts#L125

Reply:
Your right! This is already an issue starting with 4.15 when we don't saved the complete User CR anymore in the redux store. I created a bug for this so that we can backport the change (removing the `.metadata`) also to 4.15: 

https://issues.redhat.com/browse/OCPBUGS-32791 Telemetry doesn't send identify event anymore